sklearn/decomposition/_pca.py

```diff
@@ -577,9 +589,11 @@ def _fit_truncated(self, X, n_components, svd_solver):
 
         elif svd_solver == "randomized":
             # sign flipping is done inside
+            n_oversamples = int(min(X.shape) * self.n_oversamples_rate)
```

Comment:
We should validate the number with `check_scalar`. I am not sure about using a float thought.

Reply:
This will change the default behavior which currently is specified via `n_oversamples=10`.

I think a good way to handle it would be to set `n_oversamples` default value to `None` as to use `randomized_svd`'s default.